src/kbucket.rs

```diff
@@ -115,34 +119,90 @@ impl KbucketTable {
                 }
             }
         }
-        SearchResult {
-            found: false,
-            bucket_index,
-            column_index: last_empty_index,
-        }
+        Search::Failure(bucket_index, last_empty_index)
     }
 
-    fn find_bucket_index(&self, identifier: Identifier) -> usize {
+    fn xor_bucket_index(&self, identifier: Identifier) -> usize {
         let x = U256::from(self.local_node_id);
         let y = U256::from(identifier);
         let xor_distance = x ^ y;
 
-        let bucket_index = MAX_BUCKETS - (xor_distance.leading_zeros() as usize);
-        println!(
-            "Xor distance leading zeros, {}",
-            xor_distance.leading_zeros()
-        );
-        println!("Bucket index for given key: {}", bucket_index);
-        bucket_index
+        println!("Identifier: {:?}", identifier);
+        println!("Leading zeros: {}", xor_distance.leading_zeros());
+        println!("\n");
+        MAX_BUCKETS - ((xor_distance.leading_zeros() - 1) as usize)
     }
 }
 
-/// TODO:  Implement real deal tests!
-///
-/// Test find_node()      **Requires adding nodes to our table**
-/// Test search_table()
-/// Test add_node()
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::helper::testing::{mk_nodes, mk_table};
+
+    // Run "cargo test add_node -- --nocapture" to see that node was added to routing table.
```

Comment:
this is fine to leave in but it is another type of comment you wouldn't see in most code imo

Reply:
I'll put this in the justfile